tests/commands/run_test.py

```diff
@@ -88,12 +88,12 @@ def _do_run(repo, args, environ={}):
     return ret, printed
 
 
-def _test_run(repo, options, expected_outputs, expected_ret, stage):
-    if stage:
-        stage_a_file()
+def _test_run(repo, options, expected_outputs, expected_ret, file_to_stage):
```

Comment:
probably don't need this if you don't need the `files: ...` above.

I think you can still write a regression test with the bash hook, maybe not with this helper (which seems to have gotten too complicated :()

There's a few one-off tests at the bottom which this probably fits better into